### PR TITLE
Create Adaptive Platform Functional Cluster Template

### PR DESCRIPTION
**Pull request by bburda**
- Set up directory structure to the Pitchfork layout

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,150 @@
+# Created by https://www.gitignore.io/api/c++,bazel,python
+# Edit at https://www.gitignore.io/?templates=c++,bazel,python
```
Comment (insooth) [inline .gitignore:2]: https://www.gitignore.io/api/vim,c++,bazel,python

File: .clang_format [hunk 2]
```diff
@@ -0,0 +1,95 @@
+---
+Language:        Cpp
```
Comment (insooth) [inline .clang_format:2]: I do not know yet how to add this, but would be nice to indent access modifiers (like `public:`) by single space; and expose `!`  in expressions like `if ( ! something)` with spaces.

File: .gitignore (hunk 1 above)
Comment (bburda) [reply .gitignore]: Added together with Eclipse, Visual Studio Code and few others

File: .clang_format (hunk 2 above)
Comment (bburda) [reply .clang_format:2]: Could you create a new issue for this?

Comment (insooth) [reply .clang_format:2]: Done.

**Pull request merged**